### PR TITLE
radicle-httpd: Add more issue endpoint tests

### PR DESCRIPTION
**Pull request by sebastinez**
- Adds a `PATCH` method to `api::test` for e.g. updating issues.
- Adds endpoint tests for creating issues, creating a top-level comment, and replying to a comment.
- Also moves around some parts for better DX

### CODE REVIEW

**Comment by xphoniex** [comment]
lgtm, can you merge it with your test case to fix the failing lint in ci?

File: radicle-httpd/src/api/json.rs [hunk 1]
```diff
@@ -129,6 +129,7 @@ struct Author {
 #[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
 struct Comment {
+    id: OpId,
```
Comment (sebastinez) [inline radicle-httpd/src/api/json.rs:132]: I added `OpId` here to each comment, to have a reference in the web client to refer to, when needing to reply to a specific comment or to react.

File: radicle-httpd/src/api/test.rs [hunk 2]
```diff
@@ -137,37 +137,69 @@ pub fn seed(dir: &Path) -> Context {
...
+) -> Response {
+    Response(
+        app.clone()
+            .oneshot(request(path, Method::PUT, body, auth))
+            .await
+            .unwrap(),
+    )
+}
+
+fn request(
```
Comment (sebastinez) [inline radicle-httpd/src/api/test.rs:188]: I didn't want to add the `Authorization` header to every request, so I went with a conditional adding, else we pass `None` to `request`

**Comment by sebastinez** [comment]
> lgtm, can you merge it with your test case to fix the failing lint in ci?

@xphoniex, ok did add my test cases and the required changes for them to work

File: radicle-cli/tests/framework/mod.rs [hunk 3]
```diff
@@ -174,6 +174,7 @@ impl TestFormula {
                 }
                 let result = Command::new(cmd.clone())
                     .env_clear()
+                    .env("LANG", "C")
```
Comment (sebastinez) [inline radicle-cli/tests/framework/mod.rs:177]: This is to run the tests with English explicitly.
In my case e.g. I get issues showing output in german

File: radicle-cli/examples/rad-issue.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ The issue is now listed under our project.
 
 ```
 $ rad issue list
-de81d97d7fe07a80bfb339200c6af862d4526b6a "flux capacitor underpowered"
```
Comment (sebastinez) [inline radicle-cli/examples/rad-issue.md:14]: Changes to the `cob::Action` enum has changed the issue id.

File: radicle-httpd/src/api/auth.rs [hunk 5]
```diff
@@ -1,6 +1,9 @@
 use radicle::crypto::PublicKey;
 use serde::{Deserialize, Serialize};
-use time::{serde::timestamp, OffsetDateTime};
+use time::{serde::timestamp, Duration, OffsetDateTime};
```
Comment (xphoniex) [inline radicle-httpd/src/api/auth.rs:3]: we keep imports one-level deep, so instead:
```
use time::serde::timestamp;
use time::{Duration, OffsetDateTime};
```

File: radicle-httpd/src/api/v1/projects.rs [hunk 6]
```diff
@@ -542,10 +542,30 @@ async fn patch_handler(
...
     use serde_json::json;
-
-    use crate::api::test::{self, get, HEAD, HEAD_1};
+    use time::OffsetDateTime;
+
+    use crate::api::test::{self, get, patch, post, HEAD, HEAD_1};
+    use crate::api::{auth, Context};
+
+    /// Adds an authorized session to the Context::sessions HashMap.
+    async fn create_session(ctx: Context) {
```
Comment (xphoniex) [inline radicle-httpd/src/api/v1/projects.rs:554]: isn't this `create_session` fn better suited to be inside `radicle-httpd/src/api/test.rs`?

File: radicle-httpd/src/api/v1/projects.rs [hunk 7]
```diff
@@ -1021,6 +1042,203 @@ mod routes {
...
+        .unwrap();
+        let response = post(
+            &app,
+            "/projects/rad:z4FucBZHZMCsxTyQE1dfE2YR59Qbp/issues",
+            Some(Body::from(body)),
+            Some("u9MGAkkfkMOv0uDDB2WeUHBT7HbsO2Dy".to_string()),
+        )
+        .await;
+        assert_eq!(response.status(), StatusCode::CREATED);
+        assert_eq!(response.json().await, json!({ "success": true }));
```
Comment (xphoniex) [inline radicle-httpd/src/api/v1/projects.rs:1066]: can you add a space between setups/actions and asserts?
```
…
.await;

assert_eq!(response.status(), StatusCode::CREATED);
…
```

Comment (sebastinez) [reply radicle-httpd/src/api/v1/projects.rs]: 66674e4

File: radicle-httpd/src/api/v1/projects.rs (hunk 6 above)
Comment (sebastinez) [reply radicle-httpd/src/api/v1/projects.rs]: 66674e4

File: radicle-httpd/src/api/auth.rs (hunk 5 above)
Comment (sebastinez) [reply radicle-httpd/src/api/auth.rs]: 66674e4

**Pull request merged**